pandas/io/html.py

```diff
@@ -924,22 +927,22 @@ def _parse(flavor, io, match, attrs, encoding, displayed_only, **kwargs):
 
 @deprecate_nonkeyword_arguments(version="2.0")
 def read_html(
-    io,
-    match=".+",
-    flavor=None,
-    header=None,
-    index_col=None,
-    skiprows=None,
-    attrs=None,
-    parse_dates=False,
-    thousands=",",
-    encoding=None,
-    decimal=".",
-    converters=None,
+    io: FilePathOrBuffer,
+    match: Union[str, Pattern] = ".+",
+    flavor: Optional[str] = None,
+    header: Union[int, Sequence[int], None] = None,
+    index_col: Union[int, Sequence[int], None] = None,
+    skiprows: Union[int, Sequence[int], slice, None] = None,
+    attrs: Optional[Dict[str, str]] = None,
+    parse_dates: bool = False,
+    thousands: Optional[str] = ",",
+    encoding: Optional[str] = None,
+    decimal: str = ".",
+    converters: Optional[dict] = None,
```

Comment:
dict -> Dict

Reply:
In light of PEP 585 this is IMO ok?

Reply:
if there is agreement on https://github.com/pandas-dev/pandas/issues/30539#issue-543435852, this would fail CI.

Reply:
[The PEP says](https://www.python.org/dev/peps/pep-0585/#implementation):

> Importing those from typing is deprecated. Due to PEP 563 and the intention to minimize the runtime impact of typing, this deprecation will not generate DeprecationWarnings. Instead, type checkers may warn about such deprecated usage when the target version of the checked program is signalled to be Python 3.9 or newer. It's recommended to allow for those warnings to be silenced on a project-wide basis.
>
>The deprecated functionality will be removed from the typing module in the first Python version released 5 years after the release of Python 3.9.0.

i.e. importing type hints from typing (including typing.Dict) will be deprecated in 3.9. So does not look like a win to enforce that kind of type hints in pandas IMO.

Reply:
This is a good reference but at a minimum can only be a future import in Python 3.7, so I think should stick with the current convention at least until we drop 3.6

Reply:
Is there a difference between the two return signatures in 3.6? I’ve thought they’re equivalent, also in 3.6

Reply:
dict is equivalent to `Dict[Any]` or also `Dict`; obviously with dict you can't subscript until the timelines noted by the PEP you linked

We import from typing in about 99% of cases so should at least stick to that convention; if subtypes are possible here they make things even better

Reply:
Ok, I've updated the PR.